docs: replace `dvc pipeline` with `dvc dag` (#1383)
iterative/dvc.org: content/docs/command-reference/dag.md

usage: dvc dag [-h] [-q | -v] [--dot] [--full] [target]

positional arguments:
  targets         Stage or output to show pipeline for (optional)
                  Finds all stages in the workspace by default.
```

## Description

A data pipeline, in general, is a series of data processing
[stages](/doc/command-reference/run) (for example console commands that take an
input and produce an <abbr>output</abbr>). A pipeline may produce intermediate
data, and has a final result. Machine learning (ML) pipelines typically start a
with large raw datasets, include intermediate featurization and training stages,
and produce a final model, as well as accuracy
[metrics](/doc/command-reference/metrics).

In DVC, pipeline stages and commands, their data I/O, interdependencies, and
results (intermediate or final) are specified with `dvc add` and `dvc run`,
among other commands. This allows DVC to restore one or more pipelines of stages
interconnected by their dependencies and outputs later. (See `dvc repro`.)

> DVC builds a dependency graph
> ([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)) to do this.

`dvc dag` displays the stages of a pipeline up to the target stage. If `target`

> is omitted, it will show the full project DAG.

Reviewer: I think we shuold start the description with this paragraph though. Since it's now a specific command after all 🙂

Reviewer: Never mind, the short intro is enough. That one needs updating though... Will do.

## Synopsis

```usage
usage: dvc dag [-h] [-q | -v] [--dot] [--full] [target]

positional arguments:

>   targets         Stage or output to show pipeline for.
>                   Optional. (Finds all stages in the workspace by default.)

Reviewer: ```suggestion
  targets         Stage or output to show pipeline for (optional)
                  Finds all stages in the workspace by default.
```
Would be great to also update the core repo

```

## Description

`dvc dag` displays the stages of a pipeline up to the target stage. If `target`

> is omitted, it will show the full project DAG.

Reviewer: - Is it possible to have more than one DAG?
- Should we call it always "pipeline" or always "DAG" or always "dependency graph"? We currently have a mix of all these everywhere 😕

Author: Yes, 2 DAGs together is 1 DAG too, just with weakly connected components :)

Pipeline is a DAG in which all stages are somehow connected with each other, so it is not quite that. We could call it pipelineS, but `dvc add` foo is not quite a pipeline strictly speaking. `dependency graph` sounds as the most correct term, but DAG is a synonym, hence the https://dagshub.com/ :) So we could and probably should use them interchangeably.

This doc is adapted from the old `dvc pipeline show` so it does suffer from some legacy sentence structure :(

## Options

- `--dot` - show DAG in `DOT` format. It can be passed to third party

>   visualization utilities.

Reviewer: What is DOT format? Maybe link to some 3rd party doc 🙂

> - `--full` - show full DAG that the target belongs too, instead of show DAG
>   consisting only of ancestors.

Reviewer: ```suggestion
- `--full` - show full DAG that the `target` belongs too, instead of showing
  the part that consists only of the target ancestors.
```

variable. For example, the following command will replace the default pager with
[`more`](<https://en.wikipedia.org/wiki/More_(command)>), for a single run:

```dvc

> $ DVC_PAGER=more dvc dag my-pipeline.dvc

Reviewer 2: let's update it to DVC 1.0

Author: Fixed.

[`more`](<https://en.wikipedia.org/wiki/More_(command)>), for a single run:

> ```bash

Reviewer: ````suggestion
```dvc
````

## Examples

Visualize DVC pipeline:

```dvc

> $ dvc dag eval.txt.dvc

Reviewer 2: @efiop this command and output do not look correct to me for DVC 1.0

Author: Styling of the boxes is the only difference, and it was different long before 1.0. Structurally it is still correct. I would delete this for now, not to waste time on re-generating this and would take from get-started. But could just put whatever too as a placeholder for now. WIll take a look.

Reviewer 2: it's not about styling only as fr as I udnerstand

`eval.txt.dvc` should be removed?

stage names should not include `.dvc` anymore I guess

Author: @shcheklein Took a new example instead. Fixed. Thanks!
